CMakeLists.txt

```diff
@@ -0,0 +1,199 @@
+  add_definitions(-DHAS_SET_DIRECTION_ENABLED)
+endif()
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+if(PCAPPP_TARGET_PLATFORM_WINDOWS AND PCAPPP_TARGET_COMPILER_MSVC)
+  # Other hacks for VStudio
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else()
+  if(NOT CMAKE_USE_PTHREADS_INIT)
+    message(FATAL_ERROR "Pthreads not found!")
+  endif()
+endif()
+
+if(PCAPPP_USE_DPDK)
+  find_package(dpdk)
+  if(NOT dpdk_FOUND)
+    message(FATAL_ERROR "Specify DPDK Library with env DPDK_DIR=<DPDK_PATH>")
+  endif()
+  add_definitions(-DUSE_DPDK)
+endif()
+
+# Git Commit and Branch
+add_compile_definitions(GIT_COMMIT="${GIT_HASH}")
+add_compile_definitions(GIT_BRANCH="${GIT_BRANCH}")
+
+if(PCAPPP_USE_PF_RING)
+  find_package(PF_RING)
+  if(NOT PF_RING_FOUND)
+    message(FATAL_ERROR "Packet library -DPF_RING_ROOT=<PF_RING_PATH>")
+  endif()
+  add_definitions(-DUSE_PF_RING)
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release")
+endif()
+
+if(PCAPPP_TARGET_COMPILER_CLANG OR PCAPPP_TARGET_COMPILER_GCC)
+  add_compile_options(-Wall -g)
+endif()
+
+add_subdirectory(3rdParty)
+add_subdirectory(Packet++)
+add_subdirectory(Pcap++)
+add_subdirectory(Common++)
+
+if(PCAPPP_BUILD_EXAMPLES)
+  set(PCAPPP_BINARY_EXAMPLES_DIR ${CMAKE_BINARY_DIR}/examples_bin)
+  add_subdirectory(Examples)
+endif()
+
+if(PCAPPP_BUILD_TESTS)
+  include(CTest)
+  add_subdirectory(Tests)
+endif()
+
+if(PCAPPP_INSTALL)
+  set(CPACK_DEBIAN_PACKAGE_MAINTAINER "seladb")
+  set(CPACK_PACKAGE_FILE_NAME PcapPlusPlus-v${SEMVER})
+
+  include(CPack)
+
+  # Generate PKG-Config
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/PcapPlusPlus.pc.in ${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlus.pc @ONLY)
+  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlus.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+  # CMake helpers to compile Pcap++ with CMake
+  if(LIGHT_PCAPNG_ZSTD)
+    set(_PCAPPP_FIND_ZSTD "if(NOT ZSTD_FOUND)\n  find_package(ZSTD)\nendif()")
+  endif()
+
+  install(
+    EXPORT PcapPlusPlusTargets
+    DESTINATION ${PCAPP_INSTALL_CMAKEDIR}
+    NAMESPACE PcapPlusPlus::)
+
+  configure_package_config_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/PcapPlusPlusConfig.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlusConfig.cmake
+    INSTALL_DESTINATION ${PCAPP_INSTALL_CMAKEDIR})
+
+  write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlusConfigVersion.cmake
+    VERSION ${VERSION}
+    COMPATIBILITY AnyNewerVersion)
+
+  # CMake helpers to compile Pcap++ with CMake
+  install(
+    FILES ${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlusConfig.cmake
+          ${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlusConfigVersion.cmake
+          ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/FindPCAP.cmake
+          ${CMAKE_CURRENT_SOURCE_DIR}/3rdParty/LightPcapNg/LightPcapNg/cmake/FindZSTD.cmake
+    COMPONENT dev
```

Comment:
I'm not sure what `COMPONENT` means. I looked at [the documentation](https://cmake.org/cmake/help/latest/command/install.html) but it's still not clear to me

Reply:
libpcap++ runtime component are the .so e.g when an executable is compiled and required libpcap++ to be executed.
 
libpcap++ dev component are the .h / .a / .cmake use by the dev to compile the app

Reply:
Ok, that makes a lot of sense. Thanks!